### ISSUE TITLE
Error with searching for generators using search feature of yo CLI

### ISSUE THREAD

**Issue by kalkusa** [yeoman/yo#544]
## Type of issue
Bug

## My environment
* *OS version/details*: `Windows 10 Enterprise N (Azure machine)`
* *Node version:* `6.11.4` 
* *npm version:* `5.5.1` 
* *Version of yo :* `2.0.0` 

## Expected behavior
Correctly finding generators when search feature is executed. 

## Current behavior
Yeoman throws the exception when searching for npm generators is executed.

## Steps to reproduce the behavior
I've just installed Yeoman. I run `yo` command. Then from the menu I select `Install a generator`. Then Yeoman asks `Search npm for generators`. I tried a few (like `angular` or `jhipster`). Then I get the error message (see below).

## Command line output

> PS C:\Users\userName\Desktop\angularJS unit tests> yo
> ? 'Allo userName! What would you like to do? Install a generator
> ? Search npm for generators: angular
> 
>      _-----_     ╭───────────────────────╮
>     |       |    │      Bye from us!     │
>     |--(o)--|    │       Chat soon.      │
>    `---------´   │      Yeoman team      │
>     ( _´U`_ )    │    http://yeoman.io   │
>     /___A___\   /╰───────────────────────╯
>      |  ~  |
>    __'.___.'__
>  ´   `  |° ´ Y `
> 
> ParseError: Unexpected token '<' at 2:1 in https://registry.npmjs.org/-/_view/byKeyword?startkey=[%22yeoman-generator%22]&endkey=[%22yeoman-generat
> or%22,%7B%7D]&group_level=3
> <?xml version="1.0" encoding="utf-8"?>
> ^
>     at C:\Users\userName\AppData\Roaming\npm\node_modules\yo\node_modules\npm-keyword\node_modules\got\index.js:120:14
>     at BufferStream.<anonymous> (C:\Users\userName\AppData\Roaming\npm\node_modules\yo\node_modules\read-all-stream\index.js:64:3)
>     at emitNone (events.js:91:20)
>     at BufferStream.emit (events.js:185:7)
>     at finishMaybe (C:\Users\userName\AppData\Roaming\npm\node_modules\yo\node_modules\readable-stream\lib\_stream_writable.js:607:14)
>     at afterWrite (C:\Users\userName\AppData\Roaming\npm\node_modules\yo\node_modules\readable-stream\lib\_stream_writable.js:470:3)
>     at _combinedTickCallback (internal/process/next_tick.js:86:20)
>     at process._tickCallback (internal/process/next_tick.js:104:9)
> PS C:\Users\userName\Desktop\angularJS unit tests>

**Comment by SBoudrias** [comment]
Looks like an NPM ajax endpoint returned HTML instead. I'm pretty sure we don't try to parse error status code, so most likely this happened after following a redirect or simply getting a 200.

I think you should try curling this url from your server and see what is the answer you get.

Maybe a proxy is blocking access?

**Comment by sahandii** [comment]
Getting this error also

`ParseError: Unexpected token '<' at 2:1 in https://registry.npmjs.org/-/_view/byKeyword?startkey=[%22yeoman-generator%22]&endkey=[%22yeoman-generator%22,%7B%7D]&group_level=3
<?xml version="1.0" encoding="utf-8"?>
^
    at C:\usr\local\node_modules\yo\node_modules\npm-keyword\node_modules\got\index.js:120:14
    at BufferStream.<anonymous> (C:\usr\local\node_modules\yo\node_modules\read-all-stream\index.js:64:3)
    at emitNone (events.js:91:20)
    at BufferStream.emit (events.js:185:7)
    at finishMaybe (C:\usr\local\node_modules\yo\node_modules\readable-stream\lib\_stream_writable.js:607:14)
    at afterWrite (C:\usr\local\node_modules\yo\node_modules\readable-stream\lib\_stream_writable.js:470:3)
    at _combinedTickCallback (internal/process/next_tick.js:86:20)
    at process._tickCallback (internal/process/next_tick.js:104:9)`

**Comment by mischah** [comment]
Same here with node 6.10.2, npm 5.5.1 running on Mac OS 10.12.6

**Comment by mischah** [comment]
The [registry](https://registry.npmjs.org/-/_view/byKeyword?startkey=[%22yeoman-generator%22]&endkey=[%22yeoman-generator%22,%7B%7D]&group_level=3) returns:
```
<?xml version="1.0" encoding="utf-8"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN"
 "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
<html>
  <head>
    <title>503 first byte timeout</title>
  </head>
  <body>
    <h1>Error 503 first byte timeout</h1>
    <p>first byte timeout</p>
    <h3>Guru Mediation:</h3>
    <p>Details: cache-hhn1549-HHN 1509401194 2065803979</p>
    <hr>
    <p>Varnish cache server</p>
  </body>
</html>
```

https://status.npmjs.org/ says »All Systems Operational« 🤔 

I’m going to write npm support an email …

**Comment by mischah** [comment]
Feedback from npm support:

> Thanks for reaching out. I've reached out to our team. They are in the process of retiring all endpoints as there are performance issues and timeouts. Per our team, you could run the below curl command instead:
> 
>     curl -XGET https://registry.npmjs.org/-/v1/search?text=keywords:yeoman-generator | json
> 
> For more information on the new search API, please go to:
> https://github.com/npm/registry/blob/master/docs/REGISTRY-API.md#get-v1search
> 
> Please let us know if there is anything else we can do for you. We’ll be here to help. Thank you.

Must be fixed in [npm-keyword](https://github.com/sindresorhus/npm-keyword) which is used to get the generators. I’m going to create an issue over there.

**Comment by bcoe** [comment]
Just following up, you can use the search-api itself to drill down to a specific phrase, e.g., `angular` combining this with only modules that have the `yeoman-generator` keyword. To do so, structure a query like this:

`curl -XGET "https://registry.npmjs.org/-/v1/search?text=keywords:yeoman-generator+angular" | json`

This helps eliminate the need for pulling down a list of all `yeoman-generator`s, which there's a significant cost associated with in ElasticSearch. Our goal with the search-api is to get it so a relevant result is returned in a reasonably small result set, rather than returning a large result set.

If you need any help building up a query that matches your use-case, or if you have any other questions, let me know.

**Comment by mischah** [comment]
Closed with #563 and released with v2.0.1
